// vue-language-server/dist/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function getDefaultVLSConfig() {
    return {
        vetur: {
            useWorkspaceDependencies: false,
            validation: {
                template: true,
                style: true,
                script: true
            },
            completion: {
                autoImport: false,
                useScaffoldSnippets: false,
                tagCasing: 'initial',
                scaffoldSnippetSources: {
                    workspace: '💼',
                    user: '🗒️',
                    vetur: '✌'
                }
            },
            grammar: {
                customBlocks: {}
            },
            format: {
                enable: true,
                options: {
                    tabSize: 2,
                    useTabs: false
                },
                defaultFormatter: {},
                defaultFormatterOptions: {},
                scriptInitialIndent: false,
                styleInitialIndent: false
            },
            trace: {
                server: 'off'
            },
            dev: {
                vlsPath: '',
                vlsPort: -1,
                logLevel: 'INFO'
            },
            experimental: {
                templateInterpolationService: false
            }
        }
    };
}
exports.getDefaultVLSConfig = getDefaultVLSConfig;
